src/core/lib/security/credentials/tls/tls_utils.h

```diff
@@ -24,6 +24,10 @@
 #include <string>
 #include <vector>
 
+#include "openssl/evp.h"
+#include "openssl/bio.h"
+#include "openssl/x509.h"
```

Comment:
If the only call site  is `grpc_tls_certificate_provider.cc`, you probably don't want to make it in the file now. You can put everything in an anonymous namespace before `PrivateKeyAndCertificateMatch`, like:
```
namespace {
    struct EVP_PKEYDeleter {
    }
    using OwnedEVP_PKEY = ...
}  // namespace
```
Also I felt like `tls_utils.h` might not be the right place for this...originally I was considering creating a new file for helper functions to make better use of OpenSSL/BoringSSL data structures.

Reply:
It will also be used in the format checking functions. Also, I feel like a new header file works best too.